lib/Controller/RoomController.php

```diff
@@ -228,11 +234,41 @@ protected function formatRoom(Room $room, Participant $participant = null) {
 			$room->cleanGuestParticipants();
 		}
 
+		$lastMessageFromHistory = $this->chatManager->getHistory($room->getId(), 0, 1);
+		$lastMessage = '';
+
+		if (!empty($lastMessageFromHistory)) {
+
+			list($message, $messageParameters) = $this->richMessageHelper->getRichMessage($lastMessageFromHistory[0]);
+
+			$displayName = '';
+
+			if ($lastMessageFromHistory[0]->getActorType() === 'users') {
+				$user = $this->userManager->get($lastMessageFromHistory[0]->getActorId());
+				$displayName = $user instanceof IUser ? $user->getDisplayName() : '';
+			} else if ($lastMessageFromHistory[0]->getActorType() === 'guests') {
+				$guestSessions[] = $lastMessageFromHistory[0]->getActorId();
```

Comment:
You could simplify this as
```php
$guestSession = $lastMessageFromHistory[0]->getActorId();
$guestNames = !empty($guestSession) ? $this->guestManager->getNamesBySessionHashes([$guestSession]) : [];
```

You could even move the `getActorId` / 'getActorType' out of this `if` cases because you are getting them again when constructing `$lastMessage`:
```php
$actorId = $lastMessageFromHistory[0]->getActorId();
$actorType = $lastMessageFromHistory[0]->getActorType();
...
if ($actorType === 'users') {
    $user = $this->userManager->get($actorId);
    ...
) else if ($actorType === 'guests') {
    $guestNames = !empty($actorId) ? $this->guestManager->getNamesBySessionHashes($actorId) : [];
    ...
}
...
$lastMessage = [
    'actorType' => $actorType,
    'actorId' => $actorId,
];
```